compaction/compaction.cc

```diff
@@ -1364,6 +1364,12 @@ class cleanup_compaction final : public regular_compaction {
 
 class split_compaction final : public regular_compaction {
     compaction_type_options::split _options;
+
+    uint64_t partitions_per_sstable() const override {
+        // one sstable will be split into two, so use half the
+        // total estimated partitions for the output tables.
+        return _estimated_partitions / 2;
+    }
```

Comment:
A better estimate is a linear interpolation based on the token range spanned by the sstable. If the sstable spans the token range (1000, 2000) and we split at 1100, the left-hand-side would have 10% of the partitions and the right-hand side 90%.

I see partitions_per_sstable() can't give such a response, but maybe it can be adjusted.

Reply:
We can use `sstables::sstable::estimated_keys_for_range()`, it estimates partition-count for a given token-range, based on the number of summary entries that fall into said range. It should be better than linear interpolation.

Reply:
I have updated the PR to come up with a better partition estimate.